src/NUnitFramework/Directory.Build.props

```diff
@@ -3,14 +3,18 @@
   <PropertyGroup>
     <LangVersion>6</LangVersion>
     <Features>strict</Features>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\..\nunit.snk</AssemblyOriginatorKeyFile>
 
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <OutputPath>..\..\..\bin\$(Configuration)\</OutputPath>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
+    <!-- Ideally this is always enabled, but that tends to hurt developer productivity -->
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
```

Comment:
My preference would have been to wait until two properties shared the condition before adding a new property group:

```diff
    <Features>strict</Features>
+
+   <!-- Ideally this is always enabled, but that tends to hurt developer productivity -->
+   <TreatWarningsAsErrors Condition="'$(BuildingInsideVisualStudio)' != 'true'">true</TreatWarningsAsErrors>
+
+   <SignAssembly>true</SignAssembly>
```

But not important.

Reply:
I tend to treat this like a functional unit, which can be copied to other projects as needed. I'll move it if you want me to move it; let me know the decision.

Reply:
There's no problem with it, just curiosity!